### DISCUSSION TITLE
Could the factory return an instance instead of array?

### DISCUSSION THREAD

**Discussion by shawnlindstrom** [worksome/request-factories#15] [Ideas]
I have a trait I want to test directly in a unit test that is type hinted as SomethingStoreRequest. Since the request factory returns an array, I get a type mismatch error. It would be cool to have a (clean) way to get an instance of the request other than making an actual request?

For example:
```
$request = SomethingStoreRequest::factory()->create()->instance(route('something.store'), 'POST'); 

// or maybe the uri and method could be properties on the factory so it could be:
$request = SomethingStoreRequest::factory()->create()->instance(); 
```

Instead of:
```
$data = SomethingStoreRequest::factory()->create();
$request = SomethingStoreRequest::create(route('something.store'), 'POST', $data);
```

This is probably in the weeds a bit, but since it came up for me I thought I'd put it out there.

**Comment by JeffreyDavidson** [comment]
@shawnlindstrom I have a very similar situation. Were you able to find a way around this for now?

**Comment by olivernybroe** [comment]
@lukeraymonddowning got any ideas or alternative solutions for this problem? 👀

**Comment by shawnlindstrom** [comment]
@JeffreyDavidson @olivernybroe 

See my original post. You can take the array provided by the factory and create your own instance:

```
$data = SomethingStoreRequest::factory()->create();
$request = SomethingStoreRequest::create(route('something.store'), 'POST', $data);
```

Honestly, I think doing it this way is fine. I've only ever needed it once and I think if I looked closer I could probably remove the eliminate the need for it altogether.
